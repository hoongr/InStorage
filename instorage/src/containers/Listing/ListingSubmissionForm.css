.my-submission-form-date #start_date_id.DateInput_input{
    font-size:15px;
    font-family: Helvetica;
}

.my-submission-form-date #end_date_id.DateInput_input{
    font-size:15px;
    font-family: Helvetica;
}

.my-submission-form-location .location-search-input{
    width: 100%;
}

.card-container {
	margin: 0 auto;
	float: none;
	margin-bottom: 10px;
	width: 100%;
	max-width: 700px;
}
