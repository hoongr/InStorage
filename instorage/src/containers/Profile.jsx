import React from 'react';
import { withAuthorization } from '../Session';
import { Row, Card, Col } from 'react-bootstrap';
import FirebaseManager from "../api/routes/firebaseManager";
import EmailManager from "../api/routes/emailManager";
import '../models/userProfile.js';
import { withRouter } from 'react-router-dom';

import ReactDependentScript from "react-dependent-script";
import StarRatings from 'react-star-ratings';

class Profile extends React.Component {
  constructor(props) {
    super(props);
    this.state = ({
      profile: {},
      userId: "",
      fullName: "",
      venmo: "",
      phoneNum: "",
      college: "",
      bio: "",
      userType: "",
      averageRating: '',
      profilePicURL: '',
    });
  }

  componentDidMount() {
    const { auth } = this.props.firebase;
    console.log(auth.currentUser.uid);
    this.getUser(this.props.match.params.userId);
  }

  componentDidUpdate(prevProps, prevState) {
    console.log("update");
    if (prevProps.match.params.userId !== this.props.match.params.userId) {
      this.getUser(this.props.match.params.userId);
    }
  }

  getUser = userId => {
    FirebaseManager.getUserProfile(userId)
      .then(user =>
        {
          console.log(user)
          this.setState({
            profile: user,
        })
          if(typeof this.state.profile._profilePicURL === "undefined") {
            console.log("undefined pp")
            this.setState({
              profile: {
                ...this.state.profile,
                _profilePicURL: "https://cutecatsinhats-x7v0etsjgzjvirs3.netdna-ssl.com/wp-content/uploads/2016/01/Nala-the-cat-shark-hat.jpg" 
              }
            })
          }
      }
      );
  }

  render() {
    return(
      <ReactDependentScript loadingComponent={<div>icons loading...</div>} scripts={["https://use.fontawesome.com/releases/v5.11.2/js/all.js"]}>
        <div>
          <style type="text/css">
            {`
              .text-style {
                font-size: 14px;
                font-weight: bolder;
              }

              .msg-btn {
                background-color: #7CC6FE;
                color: white;
                width: 160px;
              }

              .text-box {
                width: 400px;
                margin: auto;
                font-size: 18px;
              }

              .custom-padding {
                padding-right: 35px;
              }
            `}
          </style>
          <Row className="justify-content-md-center pt-5">
            <img src={this.state.profile._profilePicURL} className="circle-profile-pic cropped-pic"/>
          </Row>
          <Row className="justify-content-md-center pt-2">
            <div className="text-center">
              <h2>{this.state.profile._name}</h2>
              <div>
                {
                  (this.state.profile._userType === 'host' || this.state.profile._userType === 'mover') && 
                  this.state.profile._averageRating !== -1 &&
                  <Row className="justify-content-md-center pb-3">
                    <StarRatings 
                      rating={this.state.profile._averageRating} 
                      starRatedColor="#274060" 
                      numberOfStars={5}
                      starDimension="25px"
                    />
                    <h5 className="pl-3">{this.state.profile._averageRating} out of 5</h5>
                  </Row>
                }
                {
                  (this.state.profile._userType === 'host' || this.state.profile._userType === 'mover') && 
                  this.state.profile._averageRating == -1 &&
                  <Row className="justify-content-md-center pb-3">
                    <StarRatings 
                      rating={0} 
                      starRatedColor="#274060" 
                      numberOfStars={5}
                      starDimension="25px"
                    />
                    <h5 className="pl-3">No Ratings</h5>
                  </Row>
                }
              </div>
              <button className="btn msg-btn btn-sm" onClick={() => {
                        this.props.history.push('/chat/' + this.state.profile._userID)}}>
                Message
              </button>
              <Row className="justify-content-md-center pt-4">
                <i class="fa fa-graduation-cap fa-2x"></i>
                <h5 className="pl-2 custom-padding">{this.state.profile._college}</h5>
                <i class="fa fa-phone fa-flip-horizontal fa-2x"></i>
                <h5 className="pl-2 custom-padding">{this.state.profile._phoneNum}</h5>
                <i class="fab fa-vimeo fa-2x"></i>
                <h5 className="pl-2">@{this.state.profile._venmo}</h5>
              </Row>
              <div className="text-box pt-3 pb-5">{this.state.profile._bio}</div>
            </div>
          </Row>
        </div>
      </ReactDependentScript>
    )
  }
}


const condition = authUser => !!authUser;

export default withAuthorization(condition)(withRouter(Profile));