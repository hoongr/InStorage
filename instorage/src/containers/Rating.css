div.dv-star-rating{
    margin-top: 16px;
}
