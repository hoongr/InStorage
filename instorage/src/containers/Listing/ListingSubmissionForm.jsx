import React, { useState } from "react";
import { Form, Col, Card, Row } from "react-bootstrap";
import "./ListingForm.css";
import { withAuthorization } from "../../Session";
import FirebaseManager from "../../api/routes/firebaseManager";
import "firebase/storage";
import 'react-dates/initialize';
import {DateRangePicker} from 'react-dates';
import 'react-dates/lib/css/_datepicker.css';
import './ListingSubmissionForm.css'
import LocationSearchInput from "../Map/LocationSearchInput";
import ReactDependentScript from "react-dependent-script";

class ListingSubmissionForm extends React.Component {
  state = {
    address: "",
    price: "",
    footage: "",
    startDate: "",
    endDate: "",
    photoUrl: "",
    userId: "",
    photo: null,
    // photoUrl: "",
    photoKey: "",
    // listing: {
    //   address: "",
    //   rentalPrice: "",
    //   squareFootageEstimate: "",
    //   dateStartAvailable: "",
    //   dateEndAvailable: "",
    //   photoURL: "",
    //   hostUserID: "",
    // },
    validated: false,
  };

  componentDidMount() {
    const { auth } = this.props.firebase;
    console.log(auth.currentUser.uid);
    this.setState({ 
      userId: auth.currentUser.uid 
    });
  }

  addListing = () => {
    console.log("called");
    const listing = {
      address: this.state.address,
      available: true,
      dateEndAvailable: this.state.endDate,
      dateStartAvailable: this.state.startDate,
      hostUserID: this.state.userId,
      rentalPrice: this.state.price,
      squareFootageEstimate: this.state.footage,
      photoURL: this.state.photoUrl
    }
    return FirebaseManager.addListing(
      listing
    );
  };

  change = e => {
    //console.log("change");
    const target = e.target;
    const value = target.type === "file" ? target.files[0] : target.value;
    const name = target.name;
    this.setState({
      [name]: value
    });
  };

  functionThatResetsTheFileInput() {
    let randomString = Math.random().toString(36);

    this.setState({
      photoKey: randomString
    });
  }

  handleSubmit = e => {
    // Input validation
    const form = e.currentTarget;
    if (form.checkValidity() === false) {
      console.log("invalid");
      e.preventDefault();
      e.stopPropagation();
      console.log(this.state);
      this.formValidation();
      this.setState({ validated: true })
      return;
    }
    else {
      console.log("valid");
      this.setState({ validated: true })
    }

    e.preventDefault();
    // Just logs fields to console
    console.log(this.state);
    const { photo } = this.state;
    const { storage } = this.props.firebase;
    const randomNum = Math.floor(Math.random() * 100000000000000);
    const uploadTask = storage
      .ref(`listings/${randomNum}${photo.name}`)
      .put(photo, { contentType: photo.type });
    uploadTask.on(
      "state_changed",
      snapshot => {
        console.log("uploading");
      },
      error => {
        console.log(error);
      },
      () => {
        this.setState({
          photoUrl: `https://firebasestorage.googleapis.com/v0/b/instorage-6b1ea.appspot.com/o/listings%2F${randomNum}${photo.name}?alt=media`
        });
        // console.log(this.state.photoUrl);
        this.addListing().then(() => {
          this.setState({
            // listing: {},
            address: null,
            price: "",
            footage: "",
            startDate: "",
            endDate: "",
            photo: null,
            photoUrl: "",
            startDateMoment: null,
            endDateMoment: null,
          });
          this.functionThatResetsTheFileInput();
          this.props.history.push("/home");
        });
        // window.location.reload();
      }
    );
  };

  formValidation() {
    var address_input = document.getElementById("address-id");
    if (!address_input.checkValidity()) {
      var x = document.createElement("STYLE");
      var t = document.createTextNode(".my-submission-form-location .location-search-input:invalid {border-color: #CC2200;}");
      var y = document.createTextNode(".my-submission-form-location .location-search-input:valid {border-color: #0BB002;}");
      x.appendChild(t);
      x.appendChild(y);
      document.getElementById("address-id").appendChild(x);
    }

    if (this.state.startDate == "") {
      var x = document.createElement("STYLE");
      var t = document.createTextNode(".my-submission-form-date #start_date_id:invalid {border-color: #CC2200;}");
      var y = document.createTextNode(".my-submission-form-date #start_date_id:valid {border-color: #0BB002;}");
      x.appendChild(t);
      x.appendChild(y);
      document.getElementById("start_date_id").appendChild(x);
    }

    if (this.state.endDate == "") {
      var x = document.createElement("STYLE");
      var t = document.createTextNode(".my-submission-form-date #end_date_id:invalid {border-color: #CC2200;}");
      var y = document.createTextNode(".my-submission-form-date #end_date_id:valid {border-color: #0BB002;}");
      x.appendChild(t);
      x.appendChild(y);
      document.getElementById("end_date_id").appendChild(x);
    }

    var footageMsg = "";
    var footageInput = document.getElementById("footage").validity;
    if (footageInput.valueMissing) {
      footageMsg = "Estimated footage of space is required.";
    }
    else if (footageInput.patternMismatch) {
      footageMsg = "Estimated footage must be a number.";
    }
    document.getElementById("footageError").innerHTML = footageMsg;

    var priceMsg = "";
    var priceInput = document.getElementById("price").validity;
    if (priceInput.valueMissing) {
      priceMsg = "Price is required.";
    }
    else if (priceInput.patternMismatch) {
      priceMsg = "Price must be a number.";
    }
    document.getElementById("priceError").innerHTML = priceMsg;
  }

  render() {
    return (
        <ReactDependentScript loadingComponent={<div>map loading...</div>} scripts={["https://maps.googleapis.com/maps/api/js?key=AIzaSyAjOpk-lRnAG_ItuCpJDU8Bo5lgqUKhSr8&libraries=places"]}>
        <div>
        <h1 className="title">List Your Space</h1>
        <Row className="justify-content-md-center">
          <Card className="card-container">
            <Card.Body>
              <Form className="form-padding" noValidate validated={this.state.validated} onSubmit={this.handleSubmit}>
                <Form.Row>
                  <Form.Group as={Col} controlId="address" className="my-submission-form-location">
                    <LocationSearchInput
                        myLabel={"Address"}
                        placeHolder={"Enter Your Storage Location"}
                        value={this.state.address}
                        onChange={(address, coordinates, fullInfo) => this.setState({ 
                          address: address,
                          coordinates: coordinates, 
                          fullAddressInfo: fullInfo})}
                        required={true}
                        id={"address-id"}
                    >
                    </LocationSearchInput>
                  </Form.Group>
                </Form.Row>

                <Form.Row className="mt-1">             
                  <Form.Group as={Col} controlId="price">
                    <Form.Label className="mb-3">Rental Price per Week</Form.Label>
                    <Form.Control
                      name="price"
                      value={this.state.price}
                      onChange={e => this.change(e)}
                      required
                      pattern="\d+"
                    />
                    <Form.Control.Feedback type="invalid">
                      {<p id="priceError"></p>}
                    </Form.Control.Feedback>
                  </Form.Group>
                  <Form.Group as={Col} controlId="footage">
                    <Form.Label className="mb-3">Estimated Square Footage</Form.Label>
                    <Form.Control
                      name="footage"
                      value={this.state.footage}
                      onChange={e => this.change(e)}
                      required
                      pattern="\d+"
                    />
                    <Form.Control.Feedback type="invalid">
                      {<p id="footageError"></p>}
                    </Form.Control.Feedback>
                  </Form.Group>
                </Form.Row>

                <Form.Row className="mt-1">
                  <Form.Group as={Col} controlId="Date" className="my-submission-form-date">
                    <Form.Label>Rental Period</Form.Label>
                    <br/>
                    <DateRangePicker 
                      startDate={this.state.startDateMoment} 
                      startDateId="start_date_id" 
                      endDate={this.state.endDateMoment} 
                      endDateId="end_date_id"
                      onDatesChange={({startDate, endDate}) => this.setState({
                                      startDateMoment: startDate,
                                      endDateMoment: endDate,
                                      startDate: (startDate === null) ? null : startDate.toISOString().substr(0,10),
                                      endDate: (endDate === null) ? null : endDate.toISOString().substr(0,10),
                                    })} 
                      focusedInput={this.state.focusedInput}
                      onFocusChange={focusedInput => this.setState({focusedInput})}
                      required={true}
                      block={true}
                    />
                    <div className="invalid-feedback">
                      Rental period is required.
                    </div>
                  </Form.Group>
                  <Form.Group as={Col} className="my-submission-form-photo">
                    <Form.Label className="mb-2">Upload a Photo</Form.Label>
                    <Form.Control
                        type="file"
                        name="photo"
                        key={this.state.photoKey || ""}
                        onChange={e => this.change(e)}
                        required
                        id="photo" 
                    />
                    <Form.Control.Feedback type="invalid">
                      Photo of space is required.
                    </Form.Control.Feedback>
                  </Form.Group>    
                </Form.Row>

                <button
                  type="submit"
                  className="btn btn-custom btn-lg"
                >
                  Post
                </button>
              </Form>
            </Card.Body>
          </Card>
        </Row>
      </div>
        </ReactDependentScript>

    );
  }
}
const condition = authUser => !!authUser;
export default withAuthorization(condition)(ListingSubmissionForm);
