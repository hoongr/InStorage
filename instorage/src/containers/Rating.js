import React from 'react';
import {withRouter} from "react-router-dom";
import FirebaseManager from "../api/routes/firebaseManager";
import StarsRating from 'stars-rating';
import StarRatingComponent from 'react-star-rating-component';
import './Rating.css';
import {Alert, Button, Modal} from "react-bootstrap";
import { Redirect } from 'react-router-dom';

class Rating extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            score: 0,
            userid: this.props.match.params.userId,
            bookingRequestId: this.props.match.params.bookingRequestId,
            profile: {
                _name: "Null",
                _profilePicURL: "https://cutecatsinhats-x7v0etsjgzjvirs3.netdna-ssl.com/wp-content/uploads/2016/01/Nala-the-cat-shark-hat.jpg",
            },
            showAlert: false,
            alertType: "success",
            redirect: false,
            showModal: false,
            inventoryList: []
        };
    }

    setRedirect = () => {
        this.setState({
            redirect: true
        })
    }

    renderRedirect = () => {
        if (this.state.redirect) {
            return <Redirect to='/' />
        }
    }

    componentDidMount() {
        FirebaseManager.getUserProfile(this.state.userid).then((profile) => {
                this.setState({profile: profile});
                if (this.state.profile._profilePicURL === undefined || this.state.profile._profilePicURL === null) {
                    this.state.profile._profilePicURL = "https://cutecatsinhats-x7v0etsjgzjvirs3.netdna-ssl.com/wp-content/uploads/2016/01/Nala-the-cat-shark-hat.jpg"
                }
                console.log(this.state.profile);
            }
        ).catch((err) => {
                console.error(err);
            }
        );
    }

    getInventoryList() {
        FirebaseManager.getBookingRequestByID(this.state.bookingRequestId).then(br => {
            console.log(br._inventoryList)
            var inventory = br._inventoryList;
            inventory = Object.keys(br._inventoryList).map(k => br._inventoryList[k]);
            this.setState({ inventoryList : inventory })
            this.formatInventoryList(inventory)
        })
    }

    formatInventoryList(list) {
        var newList = []
        list.forEach(i => 
            newList.push(<ul>{i}</ul>)
        )

        this.setState({inventoryList : newList})
    }

    populateStar() {
        var title = ["Overall", "Safety", "Punctuality", "Helpfulness"];
        var stack = [];
        title.forEach((title) => {
            stack.push(
                <div className="d-flex justify-content-between">
                    <div className={"align-self-center"}>{title}</div>
                    <div style={{fontSize: "50px"}}>
                        <StarRatingComponent
                            name={title}
                            starCount={5}
                            onStarClick={(e) => {
                                this.setState({score: e + this.state.score})
                            }}
                        />
                    </div>
                </div>
            );
        })
        return stack;
    }

    render() {
        return (
            <div>
                {this.renderRedirect()}
                <Alert
                    show={this.state.showAlert}
                    variant={this.state.alertType}
                    className="alert-centered"
                >
                    <Alert.Heading>{"Thank You! Your feedback has been recorded!"}</Alert.Heading>
                    <div className="d-flex justify-content-center">
                        <Button
                            onClick={this.setRedirect}
                            variant="outline-dark"
                        >
                            Return To Home
                        </Button>
                    </div>
                </Alert>

                {this.state.showModal && <Modal.Dialog className='modal-centered'>
                    <Modal.Header closeButton onClick={() => this.setState({ showModal: false })}>
                        <Modal.Title>Your inventory list</Modal.Title>
                    </Modal.Header>

                    <Modal.Body>
                        <div>{this.state.inventoryList}</div>
                    </Modal.Body>

                    <Modal.Footer>
                        <Button onClick={() => this.setState({ showModal: false })} variant="secondary">Close</Button>
                    </Modal.Footer>
                </Modal.Dialog>}

                <div className="flex_col, justify-center" style={{
                    width: "500px",
                    display: "block",
                    marginLeft: "auto",
                    marginRight: "auto",
                    whiteSpace: 'nowrap',
                    fontFamily: '',
                    fontSize: '30px',
                    fontWeight: '600',
                }}>
                    <div className="d-flex justify-content-center" style={{margin: "20px"}}>
                        <p>Rate your storage experience with</p>
                    </div>

                    <div className="d-flex justify-content-center" style={{marginTop: "20px"}}>
                        <img className="circle-profile-pic" src={this.state.profile._profilePicURL} alt="me"/>
                    </div>

                    <div className="mb1rem d-flex justify-content-center">
                        {this.state.profile._name}
                    </div>
                    <Button className="mb1rem w100 d-flex justify-content-center" variant="outline-primary" onClick={() => {
                        this.getInventoryList();
                        this.setState({ showModal: true })
                    }}>Check your inventory here</Button>
                    <hr/>
                    {this.populateStar()}
                    <div className={"d-flex justify-content-end"}>
                        <button type="button" className="btn btn-primary" style={{marginTop: "10px"}} onClick={() => {
                            console.log(this.state.userid, this.state.score / 4);
                            FirebaseManager.updateUserRating(this.state.userid, this.state.score / 4);
                            this.setState({score: 0, showAlert: true});
                            this.populateStar();
                        }}>Submit feedback
                        </button>
                    </div>
                </div>

            </div>
        )
    }
}

export default withRouter(Rating);
