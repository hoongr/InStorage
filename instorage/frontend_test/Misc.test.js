import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

import SignInPage from "../src/containers/SignIn"
import RegisterPage from "../src/containers/Register"
import ListingSubmissionForm from "../src/containers/Listing/ListingSubmissionForm"
import Profile from "../src/containers/Profile"
import Rating from "../src/containers/Rating"

describe('SignIn', () => {
	const wrapper = shallow(<SignInPage />);

	test("renders without crashing", () => {
		expect(wrapper).toBeDefined();
	});

	test("should match the snapshot", () => {
	  expect(wrapper).toMatchSnapshot();
	});
})

describe('Register', () => {
	const wrapper = shallow(<RegisterPage />);

	test("renders without crashing", () => {
		expect(wrapper).toBeDefined();
	});

	test("should match the snapshot", () => {
	  expect(wrapper).toMatchSnapshot();
	});
})

describe('ListingSubmission', () => {
	const wrapper = shallow(<ListingSubmissionForm />);

	test("renders without crashing", () => {
		expect(wrapper).toBeDefined();
	});

	test("should match the snapshot", () => {
	  expect(wrapper).toMatchSnapshot();
	});
})

describe('Profile', () => {
	const wrapper = shallow(<Profile />);

	test("renders without crashing", () => {
		expect(wrapper).toBeDefined();
	});

	test("should match the snapshot", () => {
	  expect(wrapper).toMatchSnapshot();
	});
});

describe('Rating', () => {
	const wrapper = shallow(<Rating />);

	test("renders without crashing", () => {
		expect(wrapper).toBeDefined();
	});

	test("should match the snapshot", () => {
	  expect(wrapper).toMatchSnapshot();
	});
});
