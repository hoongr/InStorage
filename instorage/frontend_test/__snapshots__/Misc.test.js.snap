// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingSubmission should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`Profile should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`Rating should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`Register should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`SignIn should match the snapshot 1`] = `ShallowWrapper {}`;
